src/core/thread/key_manager.cpp

```diff
@@ -179,10 +179,19 @@ KeyManager::KeyManager(Instance &aInstance)
     , mKekFrameCounter(0)
     , mIsPskcSet(false)
 {
-    Error error = mNetworkKey.GenerateRandom();
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+    SetCryptoType(OT_CRYPTO_TYPE_USE_KEY_REFS);
+#else
+    SetCryptoType(OT_CRYPTO_TYPE_USE_LITERAL_KEYS);
+#endif
 
-    OT_ASSERT(error == kErrorNone);
-    OT_UNUSED_VARIABLE(error);
+    IgnoreError(otPlatCryptoInit());
+    IgnoreError(mNetworkKey.GenerateRandom());
+
+    if (GetCryptoType() == OT_CRYPTO_TYPE_USE_KEY_REFS)
```

Comment:
I noticed some of the new code is included whether `OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE` is enabeled or not and uses `if (GetCryptoType() == OT_CRYPTO_TYPE_USE_KEY_REFS)` check. 

Would it not add code size overhead for the cases where PSA is not being used?

Can we consider adding them under `#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE`?

Reply:
That is a good idea. And it was under a similar macro, but the consensus in the previous PR was not to have PSA calls under conditional compilation. But I can certainly revert it back.